### ISSUE TITLE
`--stdout` option throws error if dash is given indicating stdin input – breaks VSCode compatibility

### ISSUE THREAD

**Issue by seweissman** [akaihola/darker#523]
**Describe the bug**
-d/--stdout option should allow "-" to indicate stdin

**To Reproduce**
1. Run  `darker -d --stdin-filename=somefile -`
2. Darker exits with error "darker.config.ConfigurationError: Either --stdin-filename=<path> or exactly one Python source file which exists on disk must be provided when using the `stdout` option"

**Expected behavior**
Darker should run and accept stdin input in the same way that it does without the trailing dash.

**Environment (please complete the following information):**
 - OS macos
 - Python version 3.9.7
 - Darker version 1.7.2
 - Black version 22.6.0

**Additional context**
I am trying to use darker with the new [VSCode black-formatter extension](https://github.com/microsoft/vscode-black-formatter/tree/main). The extension sends stdin input and seems to expect stdout, which can be accomplished by configuring it to run with `-d`. The main piece keeping it from working is that darker doesn't like the dash indicating stdin, which is added by VSCode.

I belive that [validate_stdout_src](https://github.com/akaihola/darker/blob/e96018f086383a2dcfaab6825945fbee08daca2a/src/darker/config.py#L78C6-L78C6) is likely the issue. The `validate_stdin_src` function already supports the dash.

**Comment by seweissman** [comment]
I'm happy to attempt to open a pull request for a fix f there is agreement that this is a bug.

**Comment by dkeraudren** [comment]
On my side, darker is completely broken in vscode since the python extension removed all formating options.
Since I switched to  ms-python.black-formatter I'm unable to use darker.
I use format on save but anytime I save I got a failed to save because of conflicts, no issue when I use plain black

**Comment by seweissman** [comment]
@dkeraudren I had the same issue and I believe the reason is the the VSCode extension is expecting the output on stdout, which darker doesn't do by default. I was able to get it to work by setting the formatter path to darker and by setting darker to output to stdout with `-d`, but I ran into the bug described above. I was able to patch the darker code locally to fix the bug and then it worked.

```
  "black-formatter.path": [
    "/path/to/bin/darker"
  ],
  "black-formatter.args": [
    "-d",
  ],
```

**Comment by akaihola** [comment]
Hi @seweissman and @dkeraudren, thanks for the report and analysis! I'd love to make Darker work perfectly together with the [VSCode black-formatter extension](https://github.com/microsoft/vscode-black-formatter/tree/main) as well as other IDEs and tools.

@seweissman could you share your local patch to Darker?

**Comment by akaihola** [comment]
@Garfounkel's fix is now in PR #574. I also invited you @Garfounkel, @seweissman and @dkeraudren as collaborators to this repository. If you accept the invitation, I can assign you as reviewers for #574. I'd appreciate especially a second look at the changes I made to `README.rst`. Thank you!
